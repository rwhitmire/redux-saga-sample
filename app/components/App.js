import React, { Component } from 'react'
import { connect } from 'react-redux'
import styles from './App.css'
import img from './icon.png'
import { invalidatePosts } from '../actions'

class App extends Component {
  reload() {
    this.props.invalidatePosts()
  }

  render() {
    return (
      <div className={styles.app}>
        <div>
          <img src={img}/>
        </div>
        <button onClick={() => this.reload()}>reload</button>
        <ul>
          {this.props.posts.map(post =>
            <li key={post.id}>
              <span>{post.score}</span>
              <a href={post.url}>{post.title}</a>
            </li>
          )}
        </ul>
      </div>
    )
  }
}

const actionCreators = {
  invalidatePosts
}

const mapStateToProps = ({ posts }) => ({
  posts
})

export default connect(mapStateToProps, actionCreators)(App)
