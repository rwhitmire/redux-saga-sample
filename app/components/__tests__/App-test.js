import React from 'react'
import { createStore } from 'redux'
import renderer from 'react-test-renderer'
import App from '../App'
import rootReducer from '../../reducers'

const testStore = createStore(rootReducer)

test('App should render', () => {
  const tree = renderer.create(<App store={testStore} />).toJSON()
  expect(tree).toMatchSnapshot()
})
