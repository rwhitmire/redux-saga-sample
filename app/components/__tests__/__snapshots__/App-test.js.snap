exports[`test App should render 1`] = `
<div
  className="appClassName">
  <div>
    <img
      src="test-file-stub" />
  </div>
  <button
    onClick={[Function]}>
    reload
  </button>
  <ul />
</div>
`;
